### ISSUE TITLE
Add map gaps FAQ item

### ISSUE THREAD

**Issue by jidanni** [zamojski/TowerCollector#180]
Here is a typical railroad trip. There are these odd gaps, almost regularly, and not at it's tower handoffs either. The handoffs often occur a few dots later, but  sometimes don't occur. 

Perhaps you can analyze your train and road trips, and author an FAQ item, saying what's going on. Thanks.

![Screenshot_20221006-232755_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/194462778-a75dcf5e-2da8-4201-9bf0-143403633514.jpg)
![Screenshot_20221006-232201_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/194462808-81a961b7-a884-4b49-ad8f-0bf0f655588b.jpg)
![Screenshot_20221006-231222_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/194462824-b730be45-3b44-48e4-9db4-17667a089ed2.jpg)

And no, it's probably not related to GPS satellites. Because only once in many months, did I notice a GPS glitch on the map like this,  very rare:
![Screenshot_20221006-232348_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/194462887-46af0900-f52e-40de-80ae-703c7dcae322.jpg)

**Comment by jidanni** [comment]
Here they are all the same tower, just like it is swimming and it needs a few gasps of air every few meters. A mystery. Samsung A13 5G.
![Screenshot_20221031-204930_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/199141459-8f7d8aac-1c1c-4fdd-8c8a-87998b9cfd25.jpg)![Screenshot_20221031-204903_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/199141472-7a890ecc-8fa8-4854-b5e6-36048af5cbec.jpg)![Screenshot_20221031-204858_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/199141481-47a38880-6fc7-4639-881e-9aad47807135.jpg)![Screenshot_20221031-204850_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/199141487-3336e3c2-1711-4406-b002-89e8f6f08e94.jpg)![Screenshot_20221031-204838_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/199141495-cddbeeac-358f-4e61-8919-2c280c7e0ace.jpg)

**Comment by zamojski** [comment]
I didn't observe issues like this on any of my phones. The only way to diagnose is to review the logs from the app gathered at detailed level. If you share some by email I will take a look.

**Comment by jidanni** [comment]
I'm going on a long trip (which is the perfect time to gather such stretches) at the end of the month. So what do I do?

Do I set this to debug? And then CSV files get bigger and I send you one of those?
![Screenshot_20221103-115533_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/199785820-4c1bd3f0-0513-40e0-ae87-1d6604de1197.jpg)

Or maybe the reports you want are sent automatically by email according to this setting?:
![Screenshot_20221103-115619_Tower Collector.jpg](https://user-images.githubusercontent.com/1230959/199785997-f8d6bb53-10b3-44b0-b5f8-f0709f525f46.jpg)

**Comment by zamojski** [comment]
You'll need to send the log file by email. I suggest to compress it first which should significantly reduce the file size.

The second option only reports then application crashes and doesn't include the execution details.

**Comment by jidanni** [comment]
OK I am analyzing the logs! Here is the picture:
![Screenshot_20221206-161738_Tower Collector](https://user-images.githubusercontent.com/1230959/211230882-e1332072-ff65-45c3-93e8-bc12bc3a19f2.jpg)
Comments coming soon.

**Comment by jidanni** [comment]
Long (A), and short (B, C) gap repeating pattern, https://en.m.wikipedia.org/wiki/Yucca,_Arizona .
![Screenshot_20231021_065206_Tower Collector.jpg](https://github.com/zamojski/TowerCollector/assets/1230959/b40bb19a-7ca4-45dd-8d4e-878521df1405)

**Comment by zamojski** [comment]
Without logs on debug level it's not possible to say where the issue comes from.

I'm using Tower Collector on 5 different phones and I didn't observe such behavior, so my conclusion is that's device specific problem.

**Comment by jidanni** [comment]
Yes, the higher the speed of the bus/train , the clearer the patterns. (Except I don't recall it happening with 2G GSM originating from within the same airplane.)

Anyway yes, perhaps Samsung A13 5G specific.

**Comment by zamojski** [comment]
If this is related to speed then or may be either poor GPS signal quality resulting in low accuracy or cell towers switch with a delay (at least phone delivered interrupted data).

**Comment by jidanni** [comment]
This time the train was going 90 mph instead of 79, so the gaps got stretched out so we can see them in more detail.

That part of Arizona was a perfectly flat desert with ideal cell towers.

The gaps are not related to cell tower handoff or anything like that.

Anyway indeed I did submit logs a long time ago, back in https://github.com/zamojski/TowerCollector/issues/180#issuecomment-1375019750 , please have a look.

**Comment by zamojski** [comment]
I missed that attachment, thanks! It looks like there was no update from GPS in given time but it was reconnected just before that period. It seems you have `GPS optimizations` enabled in `Collector preferences` within the app. Please turn it off and see if that helps.

And one question - do gaps appear all the time or nearly to places when train speed changes significantly?

**Comment by jidanni** [comment]
OK, I turned it off and next trip I'll see if it will help.

The gaps show up best when the train is traveling at a high constant speed in a straight line.

**Comment by jidanni** [comment]
"GPS optimizations" seems to be the critical item.

Turning them off seems to fix the problem for me.

Therefore there should be a help item for it.

It should list the pros and cons.

Saying why one should pick it, and why one might not want to pick it.

As far as the pros, well I'm not sure what they are. Maybe it is for saving electricity. Be sure to mention them. 

As far as the cons go, mention that from around 50 to 100 miles per hour, gaps appear on the map every few miles, on e.g., some Samsung phones. 

As far as why gaps do not appear at commercial airline speeds, I suppose that the plane is going so fast that distance is no longer a factor, and then the one second time check becomes the factor. And each time the time check occurs the distance is already beyond the minimum needed and a point is recorded.

As far as high speed rail bullet trains speeds are concerned, I didn't test them.

**Comment by jidanni** [comment]
@zamojski Hope you got my latest comment.

**Comment by jidanni** [comment]
Yes, that PR is to help people who see these gaps,
![Uploading 20240113T104926.jpg…]()

**Comment by zamojski** [comment]
The solution should be to validate if it still makes sense to use GPS optimization on modern devices and why it causes such issues. When I will be working on a new version I will take a closer look at it.

**Comment by jidanni** [comment]
Indeed, looking at https://github.com/barbeau/gpstest , I see 
![Screenshot_20240205_163741_GPSTest.jpg](https://github.com/zamojski/TowerCollector/assets/1230959/4e77536a-1113-4194-b379-d449b7adb517)
